templates/clientClient.gotmpl

```diff
@@ -82,7 +88,30 @@ func (a *Client) {{ pascalize .Name }}(params *{{ pascalize .Name }}Params{{ if
 		opt(op)
 	}
   {{ $length := len .SuccessResponses }}
-  {{ if .SuccessResponse }}result{{else}}_{{ end }}, err := a.transport.Submit(op)
+  timeout := utils.GetTimeout(a.cfg.RetryTimeout)
```

Comment:
nit
we could move this code to a wrapper function that gets the configuration and the submit function as arguments

Reply:
Moved the retries to the transport!